### PR TITLE
Preimport fixes

### PR DESCRIPTION
**Pull request by EliAndrewC**
High-priority changes which need to happen before we can import last year's staffers.

### CODE REVIEW

File: development-defaults.ini [hunk 1]
```diff
@@ -10,10 +10,10 @@ consent_form_url = "http://magfest.org/minorconsentform"
 regdesk_email_signature = ''' - Victoria Earl,
 MAGFest Registration Chair'''
```
Comment (EliAndrewC) [inline development-defaults.ini]: This is a semi-related change which ought to be made in puppet... but this config option isn't puppetized.  Since the config stuff is changing when we move to Docker, I went ahead and just changed it here rather than putting in the work to puppetize yet another config option.

File: uber/automated_emails.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 class AutomatedEmail:
     instances = OrderedDict()
```
Comment (EliAndrewC) [inline uber/automated_emails.py]: This is the change for #1271 so that emails require approval by default.  As you can see below, we explicitly mark a couple of emails as not needing approval, mostly the payment confirmation emails.  Other than that we'll need to be explicit about what we send, which seems like a good thing.

File: uber/automated_emails.py [hunk 3]
```diff
@@ -72,8 +72,8 @@ def __init__(self, subject, template, filter, **kwargs):
 
 
 class GuestEmail(AutomatedEmail):
```
Comment (EliAndrewC) [inline uber/automated_emails.py]: Since ``needs_approval=True`` is now already set to ``True`` there's no need to make it true in the subclasses.

File: uber/templates/preregistration/preregbase.html [hunk 4]
```diff
@@ -39,7 +39,9 @@
                 onClick: function () {
                     $('.group_fields').addClass('hide');
                     $('input[name="badge_type"]').val('{{ c.ATTENDEE_BADGE }}');
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: This is the fix for most of the problems listed in #1261 - because there was a Javascript error on the ``confirm`` form, ``togglePrices()`` was never being called.

File: uber/templates/preregistration/preregbase.html [hunk 5]
```diff
@@ -69,7 +71,7 @@
                 extra: 0
             }]
         };
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: We always want to show the tshirt row for staffers and not make them pay extra.  I considered putting this into the MagClassic repo instead of in core, but I imagine that most events provide tshirts for their staffers, so this seems like a sensible default.

File: uber/templates/regform.html [hunk 6]
```diff
@@ -78,7 +78,11 @@
     <script type="text/javascript">
         var donationChanged = function () {
             setVisible('.affiliate-row', $.val('amount_extra') > 0);
```
Comment (EliAndrewC) [inline uber/templates/regform.html]: Always show the tshirt row for people with staff badges.  Again, seems like a sensible enough default to belong in the core repo.

File: uber/templates/preregistration/preregbase.html [hunk 7]
```diff
@@ -64,12 +66,12 @@
             row: '#badge-types',
             selector: '.badge-type-selector',
             options: [{
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: Brent pointed out that on the confirmation page people might worry they weren't staffers if it just said "Attending" at the top.  So now for staff badges it says "Staffer" and "Attending" otherwise, which seems like a sensible default.

Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: I'd greatly prefer that we use the `ribbon_and_or_badge` property for this, as it's exactly what I made it for. That's an easy fix to make later, though.

File: development-defaults.ini (hunk 1 above)
Comment (binary1230) [inline development-defaults.ini]: yea, I'm fine with this.

File: uber/automated_emails.py (hunk 2 above)
Comment (binary1230) [inline uber/automated_emails.py]: very cool, I like this.

File: uber/templates/preregistration/preregbase.html (hunk 5 above)
Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: Hm, wait a minute, I'm not sure I like this. What if a staffer wants to pay for a second shirt? We put in logic last year explicitly to handle people who were entitled to two shirts, so I expect that is a relative common use-case.

**Comment by binary1230** [comment]
+1

File: uber/templates/regform.html (hunk 6 above)
Comment (kitsuta) [inline uber/templates/regform.html]: Why not just use the `gets_shirt` property here?

File: uber/templates/preregistration/preregbase.html (hunk 7 above)
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: Well I deliberately ONLY wanted it to say "Staffer" for people who actually have a staff badge, as opposed to just people with a volunteer ribbon, so I don't think ``ribbon_and_or_badge`` would be appropriate here.

Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: Ribbon_and_or_badge displays exactly what it says on their ribbon - so Volunteers would see "Volunteer," Dealers would see "Dealer," etc.

File: uber/templates/preregistration/preregbase.html (hunk 5 above)
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: I can go back and check, but to my recollection, most of the people who got two shirts were people who kicked in for the supporter package.  That will still work, this change just prevents someone from kicking in at exactly the "Add a Tshirt" level.

Here's my thinking on why this is a good change:
- If we leave that in as a donation tier for staffers, many people will assume that they need to pay an extra $20 to get a tshirt.
- If we take it out as a donation tier for staffers, some small number of staffers who understand they already get one shirt will be unable to use this form to pre-order a second shirt.

To my mind, situation 2 is preferable to situation 1, especially since they can buy a shirt on-site from the merch booth (though obviously preordering is preferable since then you're guaranteed to have the shirt available in your size when you show up to the merch booth).

Does that all sound reasonable?

File: uber/templates/regform.html (hunk 6 above)
Comment (EliAndrewC) [inline uber/templates/regform.html]: Oh, that would actually be much better!

File: uber/templates/preregistration/preregbase.html (hunk 5 above)
Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: Yup, that's fine. Do we want to change this to check the `gets_free_shirt` property instead of checking for a staff badge?

File: uber/templates/preregistration/preregbase.html (hunk 7 above)
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: Hmm, I do like that idea for the confirm page, but this might create some confusing situations for the main prereg form.  Suppose we get the following situation:
- Someone starts filling out the prereg form; they leave the default "Attendee" badge type selected
- They click the "Sign me up" box for volunteering and submit the form
- Because they didn't fill out all of the required fields, they're taken back to the form, but now the badge type option says "Volunteer"

The same thing can happen if someone clicks "Edit" before paying.

I do like the idea of showing more specific info about their registration on other places though, so maybe we can change this to say like
``{% if c.PAGE_PATH == '/preregistration/form` %}Attending{% else %}{{ attendee.ribbon_and_or_badge }}{% endif %}``

How does that sound?

Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: Sounds good! I'll do that now.

File: uber/templates/preregistration/preregbase.html (hunk 5 above)
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: That sounds good.  There's another edge case I thought of involving this change, which I just filed at #1274 but my opinion is that we don't need to hold up this PR for that.  (Though if you disagree and/or feel led to test it out now then that's fine too.)

File: uber/templates/preregistration/preregbase.html [hunk 8]
```diff
@@ -64,15 +66,15 @@
...
                 description: 'Allows access to the convention for its duration. {% price_notice "Preregistration" c.PREREG_TAKEDOWN %}',
                 extra: 0
             }]
         };
-        {% if c.SHIRT_LEVEL in c.PREREG_DONATION_TIERS %}
+        {% if c.SHIRT_LEVEL in c.PREREG_DONATION_TIERS and not attendee.gets_free_shirt %}
             BADGE_TYPES.options.push({
                 title: 'Add a tshirt',
-                description: 'Attendee level plus a {{ c.EVENT_NAME }} themed t-shirt. {% price_notice "Preregistration" c.PREREG_TAKEDOWN c.SHIRT_LEVEL %}',
+                description: 'Add a {{ c.EVENT_NAME }} themed t-shirt to your registration. {% price_notice "Preregistration" c.PREREG_TAKEDOWN c.SHIRT_LEVEL %}',
```
Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: I went ahead and changed the wording here just to make it less confusing for people who aren't actually Attendees.

**Comment by kitsuta** [comment]
I've gone ahead and made the changes that @EliAndrewC and I discussed regarding the front-end stuff. I think this is good to merge, so I'll let @binary1230 merge it if he's good with it.

**Comment by binary1230** [comment]
hey listen btw if I say +1 that also can mean 'I'm fine if you merge it', that way you guys can merge stuff when you want to instead of when someone randomly gets to it.  but I'll go ahead and merge this for ya
